### PR TITLE
[MIG] [16.0] pos_payment_change

### PR DESCRIPTION
**Pull request by julenfl**
PR for migration from v15.0 to v16.0

related task: https://github.com/OCA/pos/issues/849

### CODE REVIEW

**Comment by PierrickBrun** [comment]
Hi @julenfl ,

I'm interested in migrating this module too, do you know when you will continue/finish working on this ?

I can take it from here if this fits you.

I've done a quick test and noticed the following:
 - small problem with float_compare: I've sent you a PR to fix it
 - Apparently the pos closing button/screen has disappeared (maybe this is linked to something else in my dev env)

**Comment by julenfl** [comment]
> Hi @julenfl ,
> 
> I'm interested in migrating this module too, do you know when you will continue/finish working on this ?
> 
> I can take it from here if this fits you.
> 
> I've done a quick test and noticed the following:
> 
> * small problem with float_compare: I've sent you a PR to fix it
> * Apparently the pos closing button/screen has disappeared (maybe this is linked to something else in my dev env)

Hi @PierrickBrun 

First I apologize for not answering you. I have already accepted your PR, now I am working hard with other migrations. If you want you can continue with the migration, I can accept the PR or do it as you want.

thanks

**Review by victor-champonnois — APPROVED** [review]
Functional test OK

**Comment by victor-champonnois** [comment]
Note : there is a weird behavior when clicking to edit the old payment line it opens a wizard that doesn't make sense  
image

image

The bug already exists in version 14 though, so I guess it's not required to fix it here.

**Review by victor-champonnois — COMMENTED** [review]
Code review LGTM

**Review by RodrigoBM — APPROVED** [review]

**Comment by RodrigoBM** [comment]
@legalsylvain can you merge it, thanks

File: pos_payment_change/wizards/view_pos_payment_change_wizard.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    <record id="view_pos_payment_change_wizard_form" model="ir.ui.view">
+        <field name="model">pos.payment.change.wizard</field>
+        <field name="arch" type="xml">
+            <form>
+                <group col="4">
+                    <field name="order_id" />
+                    <field name="amount_total" invisible="1" />
+                    <field name="old_line_ids" colspan="4">
+                        <tree>
+                            <field name="old_payment_method_id" />
```
Comment (legalsylvain) [inline pos_payment_change/wizards/view_pos_payment_change_wizard.xml]:
```suggestion
                            <field name="old_payment_method_id" widget="selection"/>
```

that should do the job, regarding @victor-champonnois remarks.

File: pos_payment_change/wizards/view_pos_payment_change_wizard.xml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                    <newline />
+                    <field
+                        name="new_line_ids"
+                        colspan="4"
+                        context="{'new_line_ids': new_line_ids, 'amount_total': amount_total}"
+                    >
+                        <tree editable="bottom">
+                            <field
+                                name="new_payment_method_id"
+                                options="{'no_open': True, 'no_create_edit': True}"
```
Comment (legalsylvain) [inline pos_payment_change/wizards/view_pos_payment_change_wizard.xml]:
```suggestion
                                widget="selection"
```

same here I guess. Don't you think ?

Comment (julenfl) [reply pos_payment_change/wizards/view_pos_payment_change_wizard.xml]: I have added the changes that you have requested in the PR, thanks

File: pos_payment_change/wizards/view_pos_payment_change_wizard.xml (hunk 1 above)
Comment (julenfl) [reply pos_payment_change/wizards/view_pos_payment_change_wizard.xml]: I have added the changes that you have requested in the PR, thanks

**Comment by legalsylvain** [comment]
/ocabot merge nobump

Thanks for your contribution.

**Pull request merged**